test/cmdlineTests/standard_non_utf8_filename2/output.json

```diff
@@ -0,0 +1 @@
+{"errors":[{"type":"JSONError","component":"general","severity":"error","message":"Error parsing input JSON."}]}
```

Comment:
This looks strange. I would expect also an error message here.

Reply:
This is caused because of an exception in `solidity::util::jsonPrint(Json const& _input, JsonFormat const& _format)` where `_input` seem to be a valid json object containing the error message where it throws an exception in `_input.dump(..)`.

Reply:
Added an additional handler for `Json::exception` in `StandardCompiler::compile(..)`.

Reply:
Ok, this fix is not really correct. The main problem is that the initial error message is containing the character that created the problem, so the error message contains that invalid utf8 sequence. So during the creation of the error message, in particular the creation of the json - the json contains the invalid utf8 sequence because of the error message. Dumping that is not possible, because it contains the invalid sequence. Fixing this in a proper way will led the command line tests fail, because of that invalid utf8 sequence:

```
 - standard_non_utf8_filename
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "<frozen codecs>", line 322, in decode
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xff in position 234: invalid start byte
```

The problem is that the error message that is now created with the  `Json::exception` handler is not referring to the correct position of the input json, but it refers to the position of the error json that also contained the invalid utf8 sequence.

However, I would say we should just leave it like this, because this message at least contains the cause of the parsing problem - the invalid utf8 sequence. The clean way might be to replace all non-ascii characters with its hex-code.

Reply:
hmm... lets do it the clean way...

Reply:
Added this to [`solidity::util::jsonParseStrict(..)`](https://github.com/ethereum/solidity/blob/87d86bfba64d8b88537a4a85c1d71f521986b614/libsolutil/JSON.cpp#L229)